File: src/main/kotlin/io/titandata/titan/providers/kubernetes/Start.kt
/*
 * Copyright (c) 2019 by Delphix. All rights reserved.
 */

package io.titandata.titan.providers.kubernetes

import io.titandata.client.apis.RepositoriesApi
import io.titandata.titan.clients.Docker
import io.titandata.titan.clients.Kubernetes
import io.titandata.titan.utils.CommandExecutor

class Start (
        private val kubernetes : Kubernetes = Kubernetes(),
        private val repositoriesApi : RepositoriesApi = RepositoriesApi()
) {
    fun start(repoName: String) {
        val repo = repositoriesApi.getRepository(repoName)
        println("Updating deployment")
        kubernetes.startStatefulSet(repoName)
        println("Waiting for deployment to be ready")
        kubernetes.waitForStatefulSet(repoName)
        if (repo.properties["disablePortMapping"] != true) {
            println("Starting port forwarding")
            kubernetes.startPortForwarding(repoName)
        }
    }
}

File: internal/app/providers/kubernetes/Start.go
package kubernetes

import (
	"fmt"
	"strconv"
)

func Start(repoName string, port int) {
	cfg.BasePath = "http://localhost:" + strconv.Itoa(port)

	repo, _, _ := repositoriesApi.GetRepository(ctx, repoName)
	fmt.Println("Updating deployment")
	k8s.StartStatefulSet(repoName)

	fmt.Println("Waiting for deployment to be ready")
	k8s.WaitForStatefulSet(repoName)

	if !repo.Properties["disablePortMapping"].(bool) {
		fmt.Println("Starting port forwarding")
		k8s.StartPortForwarding(repoName)
	}
}
